### PR TITLE
UIDATIMP-783: Data Import Field Mapping Profile View: Invoice and Invoice line from EDIFACT Invoice

### PR DESCRIPTION
**Pull request by IhorKoba**
## Purpose

To display the details of the Invoice field mapping profile that was created (same general format as all other field mapping profiles)

## Approach

- Add helper functions`renderAmountValue ` and `renderCheckbox ` to`utils.js`
- Create new `MappingInvoiceDetails.js` component and related components for each section
- Fix issue with editing adjastemts and fund distribution

## Refs

https://issues.folio.org/browse/UIDATIMP-783

## Screenshots

image
image
image

### CODE REVIEW

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const folioInvoiceNumberValue = getFieldValue(mappingDetails, 'folioInvoiceNo', 'value');
+  const paymentMethodValue = getFieldValue(mappingDetails, 'paymentMethod', 'value');
+  const checkSubscriptionOverlapValue = getFieldValue(mappingDetails, 'chkSubscriptionOverlap', 'booleanFieldAction');
+  const exportToAccountingValue = getFieldValue(mappingDetails, 'exportToAccounting', 'booleanFieldAction');
+  const currencyValue = getFieldValue(mappingDetails, 'currency', 'value');
+  const currentExchangeRateValue = getFieldValue(mappingDetails, 'currentExchangeRate', 'value');
+  const setExchangeRateValue = getFieldValue(mappingDetails, 'exchangeRate', 'value');
+
+  const checkSubscriptionOverlap = renderCheckbox('invoice.extendedInformation.field.checkSubscriptionOverlap', checkSubscriptionOverlapValue);
+  const exportToAccounting = renderCheckbox('invoice.extendedInformation.field.exportToAccounting', exportToAccountingValue);
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js:35]:
```suggestion
  const checkSubscriptionOverlapCheckbox = renderCheckbox('invoice.extendedInformation.field.checkSubscriptionOverlap', checkSubscriptionOverlapValue);
  const exportToAccountingCheckbox = renderCheckbox('invoice.extendedInformation.field.exportToAccounting', exportToAccountingValue);
```

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            value={folioInvoiceNumberValue || prohibitionIconElement('folio-invoice-number')}
+          />
+        </Col>
+        <Col
+          data-test-payment-method
+          xs={3}
+        >
+          <KeyValue
+            label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.extendedInformation.field.paymentMethod`} />}
+            value={paymentMethodValue || noValueElement}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js:58]: The `KeyValue` component checks whether the value exists or not and if not, then renders the `NoValue` component on its own, so you do not need this check.
```suggestion
            value={paymentMethodValue}
```

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            value={currentExchangeRateValue || prohibitionIconElement('current-exchange-rate')}
+          />
+        </Col>
+        <Col
+          data-test-exchange-rate
+          xs={3}
+        >
+          <KeyValue
+            label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.extendedInformation.field.setExchangeRate`} />}
+            value={setExchangeRateValue || noValueElement}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js:105]:
```suggestion
            value={setExchangeRateValue}
```

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceAdjustments.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        key={index}
+      >
+        <Row left="xs">
+          <Col
+            data-test-invoice-adjustment-description
+            xs={3}
+          >
+            <KeyValue
+              label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.invoice.invoiceAdjustments.field.description`} />}
+              value={descriptionValue || noValueElement}
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceAdjustments.js:111]:
```suggestion
              value={descriptionValue}
```

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceLineInformation.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+  renderCheckbox,
+} from '../../utils';
+import { mappingProfileFieldShape } from '../../../../../utils';
+
+export const InvoiceLineInformation = ({ invoiceLineMappingDetails }) => {
+  const noValueElement = <NoValue />;
+  const prohibitionIconElement = fieldName => <ProhibitionIcon fieldName={fieldName} />;
+
+  const descriptionValue = getFieldValue(invoiceLineMappingDetails, 'description', 'value');
+  const POLineNumberValue = getFieldValue(invoiceLineMappingDetails, 'poLineId', 'value');
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceLineInformation.js:29]: Name the variable with a lowercase, please

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/VendorInformation.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  const noValueElement = <NoValue />;
+
+  const vendorInvoiceNumberValue = getFieldValue(mappingDetails, 'vendorInvoiceNo', 'value');
+  const vendorNameValue = selectedOrganization?.name;
+  const accountingCodeValue = getFieldValue(mappingDetails, 'accountingCode', 'value');
+
+  useEffect(() => {
+    if (vendorId) {
+      if (selectedOrganization.id !== vendorId) {
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/VendorInformation.js:36]:
```suggestion
    if (vendorId && selectedOrganization.id !== vendorId) {
```

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js (hunk 1 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js]: done

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js (hunk 2 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js]: done

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js (hunk 3 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/ExtendedInformation.js]: done

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceAdjustments.js (hunk 4 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceAdjustments.js]: done

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceLineInformation.js (hunk 5 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/InvoiceLineInformation.js]: done

File: src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/VendorInformation.js (hunk 6 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/detailsSections/view/InvoiceDetailsSection/VendorInformation.js]: done

**Review by mariia-aloshyna — APPROVED** [review]

**Review by IvanKryzhanovskyi — APPROVED** [review]

**Pull request merged**